rocAL/source/api/rocal_api.cpp

```diff
@@ -105,3 +105,37 @@ rocalVerify(RocalContext p_context) {
     }
     return ROCAL_OK;
 }
+
+RocalStatus ROCAL_API_CALL
+rocalSerialize(RocalContext rocal_context, size_t &serialized_string_size) {
+    auto context = static_cast<Context*>(rocal_context);
+    try {
+        context->master_graph->serialize(serialized_string_size);
+    } catch (const std::exception& e) {
+        context->capture_error(e.what());
+        ERR(e.what())
+        return ROCAL_RUNTIME_ERROR;
+    }
+    return ROCAL_OK;
+}
+
+RocalStatus ROCAL_API_CALL
+rocalGetSerializedString(RocalContext rocal_context, char* serialized_string) {
+    auto context = static_cast<Context*>(rocal_context);
+    try {
+        if (!serialized_string) {
+            THROW("String copy failed, Invalid pointer passed for serialize")
+        }
+
+        auto& serialize_pipe_string = context->master_graph->get_serialized_string();
+        if (serialize_pipe_string.empty())
+            THROW("Serialized string is empty, Invoke rocalSerialize before obtaining the string")
+        std::memcpy(serialized_string, serialize_pipe_string.c_str(), serialize_pipe_string.size() + 1);
```

Comment:
Nit: we rely on `std::memcpy` here but the TU doesn’t include `<cstring>`. MSVC and some libstdc++ configurations won’t pull that declaration in transitively, so please add the include to keep the build portable.